tools/mkcomposefs.c

```diff
@@ -893,6 +876,307 @@ static ssize_t write_cb(void *_file, void *buf, size_t count)
 	return fwrite(buf, 1, count, file);
 }
 
+static void log_message_with_time( const char* msg )
+{
+	time_t t 		= time(NULL);
+  	struct tm tm 	= *localtime(&t);
+	fprintf(stdout, "%02d:%02d:%02d\t %s\n", 
+				tm.tm_hour, tm.tm_min, tm.tm_sec, msg);
+}
+
+struct ParallelCopyInfo
```

Comment:
Really, i think it makes more sense for this to be an array instead of a linked list. With the change to a state struct to keep the current pointer that should be an easy conversion to make. There shouldn't be a need to allocate a ton of individual ParallelCopyInfo objects.

Reply:
Changed it to array